packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { once, uniqueId, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { ifCondition } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import AutosaveMonitor from '../autosave-monitor';
+
+const requestIdleCallback = window.requestIdleCallback ? window.requestIdleCallback : window.requestAnimationFrame;
+
+/**
+ * Time interval at which local autosave should occur after a change, in
+ * seconds.
+ *
+ * @var {number}
+ */
+const AUTOSAVE_INTERVAL_SECONDS = 15;
+
+/**
+ * Function which returns true if the current environment supports browser
+ * sessionStorage, or false otherwise. The result of this function is cached and
+ * reused in subsequent invocations.
+ */
+const hasSessionStorageSupport = once( () => {
+	try {
+		// Private Browsing in Safari 10 and earlier will throw an error when
+		// attempting to set into sessionStorage. The test here is intentional in
+		// causing a thrown error as condition bailing from local autosave.
+		window.sessionStorage.setItem( '__wpEditorTestSessionStorage', '' );
+		window.sessionStorage.removeItem( '__wpEditorTestSessionStorage' );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+} );
+
+/**
+ * Function returning a sessionStorage key to set or retrieve a given post's
+ * automatic session backup.
+ *
+ * Keys are crucially prefixed with 'wp-autosave-' so that wp-login.php's
+ * `loggedout` handler can clear sessionStorage of any user-private content.
+ *
+ * @see https://github.com/WordPress/wordpress-develop/blob/6dad32d2aed47e6c0cf2aee8410645f6d7aba6bd/src/wp-login.php#L103
+ *
+ * @param {string} postId  Post ID.
+ * @return {string}        sessionStorage key
+ */
+function postKey( postId ) {
+	return `wp-autosave-block-editor-post-${ postId }`;
+}
+
+/**
+ * Custom hook which returns a callback function to be invoked when a local
+ * autosave should occur.
+ *
+ * @return {Function} Callback function.
+ */
+function useAutosaveCallback() {
+	const {
+		postId,
+		getEditedPostAttribute,
+	} = useSelect( ( select ) => ( {
+		postId: select( 'core/editor' ).getCurrentPostId(),
+		getEditedPostAttribute: select( 'core/editor' ).getEditedPostAttribute,
+	} ) );
+
+	return useCallback( () => {
+		const saveToSessionStorage = () => {
+			window.sessionStorage.setItem( postKey( postId ), JSON.stringify( {
+				post_title: getEditedPostAttribute( 'title' ),
+				content: getEditedPostAttribute( 'content' ),
+				excerpt: getEditedPostAttribute( 'excerpt' ),
+			} ) );
+		};
+		requestIdleCallback( saveToSessionStorage );
+	}, [ postId, getEditedPostAttribute ] );
```

Comment:
`getEditedPostAttribute` - we put this action as a dependency here, but we don't do it for all actions in `useAutosaveNotice`. The question is why it is necessary here or why do we miss all those actions for `useEffect` inside `useAutosaveNotice`.

Is it because of `useAutosaveNotice` should be triggered only once per post id?

Reply:
Per individual chat, let's drop `getEditedPostAttribute` from the deps. We don't yet have strong conventions on how to deal with this in React hooks.

If we were very strict about this we would list every single reference used in the closure, but the references in our case are all static constant functions, either directly (defined in the module or imported) or indirectly (selectors obtained via `select`). I don't feel strongly, but I'd err toward keeping references out for now and adding them if we tighten our expectations here.